compiler/rustc_mir_build/src/build/matches/mod.rs

```diff
@@ -1742,7 +1742,6 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
         }
 
         // Deduplicate and ensure a deterministic order.
-        all_fake_borrows.sort();
         all_fake_borrows.dedup();
```

Comment:
Isn't sort+dedup different than just dedup? i.e. don't we need to do something like above with `.retain(|borrow| dedup.insert(borrow))`?